packages/curve-ui-kit/src/features/connect-wallet/lib/wagmi/wagmi-config.ts

```diff
@@ -63,14 +67,19 @@ export const createWagmiConfig = memoize(
            *    - Limited RPC method support
            *    - Rate-limiting of wallet provider
            *
-           * 2. http transport - Used as fallback when connector transport fails
-           *    - Configured with batch size of 3 to comply with DRPC free tier limits
+           * 2. Multiple http transports - Used as fallback when connector transport fails
+           *    - Primary: http transport with default URL (batch size 3 for DRPC compliance)
+           *    - Fallbacks: Additional http transports for each fallback RPC URL
            *    - Prevents error code 29 from exceeding batch limits
            *
-           * Note: WalletConnect ignores this http endpoint and uses chain.rpcUrls.default.http[0]
-           * regardless of the configuration here.
+           * Note: WalletConnect ignores this transport configuration and uses chain.rpcUrls.default.http
+           * in order, but having multiple transports helps with injected wallet resilience.
            */
-          fallback([unstable_connector(injected), http(undefined, { batch: { batchSize: 3 } })]),
+          fallback([
+            unstable_connector(injected),
+            http(undefined, { batch: { batchSize: 3 } }),
+            ...(RPC[id] ?? []).map((url) => http(url, { batch: { batchSize: 3 } })),
```

Comment:
```suggestion
...([undefined, ...RPC[id] ?? []]).map((url) => http(url, { batch: { batchSize: 3 } })),
```
freestyling from github

Reply:
undefined is weird, I've added all the different URLs instead and not as the first one